### ISSUE TITLE
Meson configure incorrectly detects -Qunused-arguments

### ISSUE THREAD

**Issue by lancethepants** [gsliepen/tinc#383]
I noticed I was getting lots of warnings `unrecognized option '-Qunused-arguments'` while cross-compiling tinc. Looking in the meson logs  it is showing it is supported despite `Compiler stderr` saying that it's not recognized by the compiler.
```
Running compile:
Working directory:  /home/lance/freshtomato-tinc/release/src-rt-6.x.4708/router/tinc/build/meson-private/tmpwfsuztqf
Command line:  arm-brcm-linux-uclibcgnueabi-gcc -I/home/lance/freshtomato-tinc/release/src-rt-6.x.4708/router/zlib/staged/usr/include -I/home/lance/freshtomato-tinc/release/src-rt-6.x.4708/router/lzo/staged/usr/include -I/home/lance/freshtomato-tinc/release/src-rt-6.x.4708/router/lz4/staged/usr/include -I/home/lance/freshtomato-tinc/release/src-rt-6.x.4708/router/openssl-1.1/staged/usr/include /home/lance/freshtomato-tinc/release/src-rt-6.x.4708/router/tinc/build/meson-private/tmpwfsuztqf/testfile.c -o /home/lance/freshtomato-tinc/release/src-rt-6.x.4708/router/tinc/build/meson-private/tmpwfsuztqf/output.obj -c -DLINUX26 -DCONFIG_BCMWL5 -DCONFIG_BCMWL6 -DCONFIG_BCMWL6A -DPART_JFFS2_GAP=0UL -pipe -fno-strict-aliasing -DBCMWPA2 -DBCMARM -fno-delete-null-pointer-checks -marm -DTCONFIG_NVRAM_64K -DLINUX_KERNEL_VERSION=132644 -ffunction-sections -fdata-sections -D_FILE_OFFSET_BITS=64 -O0 -Qunused-arguments

Code:
 extern int i;
int i;

Compiler stdout:

Compiler stderr:
 arm-brcm-linux-uclibcgnueabi-gcc: unrecognized option '-Qunused-arguments'

Compiler for C supports arguments -Qunused-arguments: YES```

**Comment by hg** [comment]
Could you try simply removing it? It was used to silence clang warnings about flag combinations that are not relevant anymore IIRC.

https://github.com/hg/tinc/actions/runs/2286199075

**Comment by eli-schwartz** [comment]
> Looking in the meson logs it is showing it is supported despite `Compiler stderr` saying that it's not recognized by the compiler.

If you try running this compiler on the command line by hand, what does it say when given that option?

```
arm-brcm-linux-uclibcgnueabi-gcc: unrecognized option '-Qunused-arguments'
```

For me, I get:
```
gcc: error: unrecognized command-line option ‘-Qunused-arguments’
```

and the compiler exits 1 for failure without producing compiled output.

Why are you getting lots of warnings instead of lots of errors?

**Comment by gsliepen** [comment]
Fixed by PR #384.
